src/Social/Contract/Comments/CommentsTrait.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Kanvas\Packages\Social\Contract\Comments;
+
+use Phalcon\Http\Response;
+use Kanvas\Packages\Social\Services\Comments;
+use Gewaer\Models\MessageComments;
+use Baka\Validation as CanvasValidation;
+use Phalcon\Validation\Validator\PresenceOf;
+
+/**
+ * Channels Trait
+ */
+trait CommentsTrait
+{
+    /**
+     * Get all the comments
+     *
+     * @param int $messageId
+     * @throws Exception
+     *
+     * @return Response
+     */
+    public function getAllComments(int $messageId) : Response
+    {
+        return $this->response($this->processOutput(MessageComments::findOrFail([
+            "conditions" => "message_id = :message_id: and apps_id = :apps_id: and companies_id = :companies_id: and is_deleted = 0",
+            "bind" => [
+                "message_id" => $messageId,
+                "apps_id" => $this->app->getId(),
+                "companies_id" =>$this->userData->getCurrentCompany()->getId(),
+            ]
+        ])));
+    }
+
+    /**
+     * Get comment by its id
+     *
+     * @param int $commentId
+     * @return Response
+     */
+    public function getComment(int $commentId): Response
+    {
+        return $this->response(Comments::getById((string)$commentId));
```

Comment:
@rwhite27 dont we have to check that we have access to this comment by checking company and the other validations?

Reply:
true true. Will add those validations